packages/jsii/lib/assembler.ts

```diff
@@ -1175,6 +1162,28 @@ function _fingerprint(assembly: spec.Assembly): spec.Assembly {
   return { ...assembly, fingerprint };
 }
 
+/**
+ * Removes the `Optional` modifier from parameters that have it, but are not followed only with
+ * other `Optional` or `Variadic` parameters. This is done after parsing out all parameters as it
+ * is otherwise difficult to assert whether a parameter with a default value is `Optional` or it it
+ * merely has a `nullable` `#type`.
+ */
+function _removeInvalidOptionalModifiers(parameters?: spec.Parameter[]) {
+  if (!parameters) {
+    return;
+  }
+
+  for (const param of parameters.filter(_cannotBeOptional)) {
+    delete param.modifier;
+  }
+
+  function _cannotBeOptional(param: spec.Parameter, index: number, array: spec.Parameter[]): boolean {
+    return spec.isOptional(param)
+      // Checks whether there is one non-optional/variadic parameter after this one in the array
+      && array.find((p, i) => i > index && !spec.isOptional(p) && !spec.isVariadic(p)) != null;
```

Comment:
Can be `array.some()`

Reply:
Ugh I never find it because I keep thinking about the ruby name... `.any?`